hook-tests/602-create-dpkg-yaml.test

```diff
@@ -0,0 +1,256 @@
+#!/bin/sh -e
+#
+# test for hook that creates dpkg.yaml used for OSS compliance
+
+#first check if file exists
+FILE=usr/share/snappy/dpkg.yaml
+test -e $FILE
+
+#second check the PPA section of dpkg.yaml
+ppa_section=$(cat <<EOF
+package-repositories:
+- type: apt
+  ppa: canonical-foundations/ubuntu/ubuntu-image
+EOF
+)
+
+if grep -Fxq "$ppa_section" $FILE
+then
+  echo "Content of PPA section is as expected in dpkg.yaml."
+else 
+  echo "Error! PPA section of dpkg.yaml not as expected."
+  echo "If that is intentional, please update the ppa_section content in the"
+  echo "test file: core18/hook-tests/602-create-dpkg-yaml.test."
+  exit 1
+fi
+
+#third check the Packages section of dpkg.yaml
+packages_section=$(cat <<EOF
+packages:
+- adduser=3.116ubuntu1
```

Comment:
Adding a test that checks for these exact package versions seems risky (unless I miss something) - the core18 snap will get new versions because of e.g. security updates and then this test will start failing